app/code/community/Elgentos/CodebaseExceptions/Model/Log/Writer/Stream.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+class Elgentos_CodebaseExceptions_Model_Log_Writer_Stream extends Zend_Log_Writer_Stream
+{
+    /**
+     * Write a message to the log.
+     *
+     * @param  array  $event  event data
+     * @return void
+     * @throws Zend_Log_Exception
+     */
+    protected function _write($event)
```

Comment:
This file replaces default implementation of the Log Stream Writer used by Magento.
(by default Magento uses  Zend_Log_Writer_Stream- the class I'm inheriting from). 
I'm replacing the default writer by configuration in the config.xml. See https://github.com/airbrake/Airbrake-Magento/pull/4/files#diff-d34b2ab1e300b8a1010535cec8f63500R65
This way I can push stuff to Airbrake API. 
So thanks to this patch all exception which has been caught and logged into exception.log will also go to Airbrake. This is super useful as in most cases Magento will not throw exception to the user, but silently log it.